flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/util/AWSUtil.java

```diff
@@ -50,8 +50,14 @@ public static AmazonKinesisClient createKinesisClient(Properties configProps) {
 		awsClientConfig.setUserAgent("Apache Flink " + EnvironmentInformation.getVersion() +
 			" (" + EnvironmentInformation.getRevisionInformation().commitId + ") Kinesis Connector");
 
-		AmazonKinesisClient client =
-			new AmazonKinesisClient(AWSUtil.getCredentialsProvider(configProps).getCredentials(), awsClientConfig);
+		AWSCredentialsProvider credentialsProvider = AWSUtil.getCredentialsProvider(configProps);
+
+		if (credentialsProvider != null) {
+			AmazonKinesisClient client =
+				new AmazonKinesisClient(credentialsProvider.getCredentials(), awsClientConfig);
+		} else {
+			AmazonKinesisClient client = new AmazonKinesisClient(awsClientConfig);
+		}
```

Comment:
Well that's embarassing - I was so sure I compiled this first! That's what I get for pushing a pull request late on a Friday.